lib/AST/ModuleNameLookup.cpp

```diff
@@ -19,9 +19,12 @@ using namespace swift;
 using namespace namelookup;
 
 namespace {
-  using ModuleLookupCache = llvm::SmallDenseMap<ModuleDecl::ImportedModule,
-                                                TinyPtrVector<ValueDecl *>,
-                                                32>;
+  // The usable results of a lookup in a particular module may differ based on
+  // where the lookup is happening.
+  using ModuleLookupCacheKey = std::pair<ModuleDecl::ImportedModule,
+                                         const DeclContext * /*lookupScope*/>;
+  using ModuleLookupCache = llvm::SmallDenseMap<ModuleLookupCacheKey,
+                                                TinyPtrVector<ValueDecl *>, 32>;
```

Comment:
I've always wondered where the sizes of the Small collections come from. Is 32 just a guess, or is there some performance cliff if it were to change to 16? It's one of those small distractions in reading the code for me which can add up over time. Is the expectation that most apps have < 32 modules? Are there measurements?